build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/HeadlessBuilder.java

```diff
@@ -369,6 +371,10 @@ protected int importProject(String projURIStr, boolean recurse) throws CoreExcep
 				}
 			}
 
+			// do not go into special directories
```

Comment:
Can this check be moved into for loop a few lines down so it only applies to recursed directories?

Reply:
I was thinking the case of import as well where user can give regex matching `.metadata`.. should we not ignore in case of import?

Reply:
Sure - lets leave it where you have it.